libmachine/persist/secretstore.go

```diff
@@ -0,0 +1,203 @@
+package persist
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rancher/machine/libmachine/host"
+	"github.com/rancher/machine/libmachine/log"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1types "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type SecretStore struct {
+	Store
+	SecretName, SecretNamespace string
+	SecretClient                corev1types.SecretInterface
```

Comment:
Hmm secret Interface vs Lister, for `gets` ideally we always use lister for cache purposes, but since this is a 1 time operation maybe not so bad. It really depends on whole system, who is reading/writing to that secret. If you provision 1k nodes this still only gets saved once right.

Reply:
Right now, if a create is successful, the secret will be read twice and written to at least twice by rancher-machine.

Reply:
This is prob fine for now unless we start to see performance issues.